### DISCUSSION TITLE
Writing pandas DataFrame with Uproot5, var * var * float64

### DISCUSSION THREAD

**Discussion by brightneedle** [scikit-hep/uproot5#945] [Q&A]
I have a dataframe where one or more columns are lists of 1D arrays, eg:
```
   run_number  event_number  pass_vbf_sel  ...                                        bjetTrigSFs      truth_mhh ntag
0      310691          1593         False  ...  ((0.9997807, 0.9997807, 0.9997807, 0.9997807, ...  719639.263731    3
1      303208          1353          True  ...  ((0.9832633, 0.98921436, 0.97731936, 0.9911188...  401586.494818    2
2      307259           419          True  ...  ((0.9763006, 0.98883635, 0.9638313, 0.98624736...  714343.701542    4
3      300863          1293         False  ...  ((0.99956995, 0.99956995, 0.99956995, 0.999569...  492971.115823    3
4      307259           873         False  ...  ((0.9992055, 0.9992055, 0.9992055, 0.9992055, ...  301056.351272    2
```

When I try and write the dataframe to .root using `uproot.recreate`, I get the following error:

```
TypeError: cannot write Awkward Array type to ROOT file:

    var * var * float64
```
Is there a way to get round this? I have to be able to keep the same structure as the dataframe, i.e. making N separate 1-D arrays won't work for my application

**Comment by jpivarski** [comment]
This is a data type that Uproot can't write. (The disk-format for `var * var * float64` is completely different from the format for `var * float64`, and `var * float64` can be vectorized, whereas `var * var * float64` can't.)
